syntax = "proto3";

message Header {
  uint32 version = 1;
  bytes  id = 2;
  uint32 seq_no = 3;
  uint64 tv_sec = 4;
  uint32 tv_usec = 5;
}

message Data {
  // Sensor data
  float temperature_c = 1;
  float pressure_mbar = 2;
  float humidity_percent = 3;

  float primary_voltage = 4;
  float solar_voltage = 5;
  float secondary_voltage = 6;
  bool  vbat_ok = 7;

  float light_color_cct_k = 8;
  uint32 light_color_red = 9;
  uint32 light_color_green = 10;
  uint32 light_color_blue = 11;
  uint32 light_color_clear = 12;
  float light_lux = 13;

  bool motion = 14;

  // System data
  string discovery = 15;
  string version = 16;

  // Thread Info
  uint32 thread_rloc16 = 30;
  bytes  thread_ext_addr = 31;
  uint32 thread_parent_avg_rssi = 32;
  uint32 thread_parent_last_rssi = 33;
}

message PermamoteMessage {
  Header header = 1;
  Data data = 2;
}
